src/classes/RD2_ScheduleService_TEST.cls

```diff
@@ -182,9 +182,265 @@ private with sharing class RD2_ScheduleService_TEST {
                 .withStatusClosed()
                 .build();
 
-        List<RecurringDonationSchedule__c> schedules = new RD2_ScheduleService().getNewSchedules(rds);
+        List<RecurringDonationSchedule__c> schedules = new RD2_ScheduleService().getNewSchedules(rd);
+
+        System.assertEquals(0, schedules.size(), 'Number of schedule records should match');
+
+    }
+
+    /***
+    * @description Verifies next donation date is not calculated for closed recurring donation.
+    */
+    @isTest
+    private static void shouldNotCalculateNextDonationDateForClosedRecurringDonation() {
+
+        npe03__Recurring_Donation__c rd =
+            TEST_RecurringDonationBuilder.constructEnhancedBuilder()
+                .withMockId()
+                .withContact(UTIL_UnitTestData_TEST.mockId(Contact.SObjectType))
+                .withAmount(400)
+                .withPaymentMethod('Check')
+                .withInstallmentPeriodMonthly()
+                .withDayOfMonth('23')
+                .withInstallmentFrequency(1)
+                .withDateEstablished(Date.newInstance(2019, 9, 15))
+                .withStartDate(Date.newInstance(2019, 11, 24))
+                .withStatusClosed()
+                .build();
+
+        System.assertEquals(null,
+            new RD2_ScheduleService().getNextDonationDate(rd),
+            'Next Donation Date should be null');
+
+    }
+
+    /***
+    * @description Verifies next donation date for monthly donation when donation falls in next month.
+    */
+    @isTest
+    private static void shouldCreateNextDonationDateInNextMonth() {
+
+        npe03__Recurring_Donation__c rd =
+            TEST_RecurringDonationBuilder.constructEnhancedBuilder()
+                .withMockId()
+                .withContact(UTIL_UnitTestData_TEST.mockId(Contact.SObjectType))
+                .withAmount(400)
+                .withPaymentMethod('Check')
+                .withInstallmentPeriodMonthly()
+                .withDayOfMonth('23')
+                .withInstallmentFrequency(1)
+                .withDateEstablished(Date.newInstance(2019, 9, 15))
+                .withStartDate(Date.newInstance(2019, 11, 24))
+                .build();
+
+        System.assertEquals(Date.newInstance(2019, 12, 23),
+            new RD2_ScheduleService().getNextDonationDate(rd),
```

Comment:
Maybe you can use a helper method that takes a date override:
```
private RD2_ScheduleService getScheduleServiceWithDateOverride(Date currDateOverride) {
    RD2_ScheduleService scheduleService = new RD2_ScheduleService();
    scheduleService.testCurrentDate = currDateOverride;
}
```

Reply:
Thanks!  That's a good thought.